pkg/api/jobrunintervals/gcsscanner.go

```diff
@@ -0,0 +1,216 @@
+package jobrunintervals
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/openshift/sippy/pkg/apis/api"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/html"
+)
+
+// Original code borrowed from https://github.com/vrutkovs/kaas/
```

Comment:
Could we not consolidate our other GCS scanning code to make it work for both?

https://github.com/openshift/sippy/tree/master/pkg/prowloader/gcs

Reply:
Wasn't aware of that code, should be possible, I'll take a look.

Reply:
Ok I gave this a go, but it bubbles up into the server now needing gcs client, gcs credentials, possibly new manifest changes for the cli args. All of this would be for code I intend to fully delete once I can get these stored in some system like opensearch/loki. I don't think switching the current gcs code to use what I added is worthwhile as the current code is probably more correct.

My request here is to leave this be, I will delete it once I can get a better source for the intervals to live and query from.

Reply:
Actually hang on, we may still need this code. Unless we decide to stuff everything into opensearch/loki soon, we're going to need something to load individual jobs we're interested. I expected that to be sippy, if we don't have the data, pull it from gcs and load it up. In that case, gcs would still be needed, and doing this more properly may make sense.

Reply:
Now switched over to use the gcs code you wrote for prow. Seems faster too.